lte/gateway/c/session_manager/SessionState.cpp

```diff
@@ -235,26 +235,64 @@ uint32_t SessionState::get_current_version() {
 }
 
 void SessionState::set_current_version(
-    int new_session_version, SessionStateUpdateCriteria* session_uc) {
-  current_version_ = new_session_version;
-  if (session_uc) {
-    session_uc->is_current_version_updated = true;
-    session_uc->updated_current_version    = new_session_version;
-  }
+    int new_session_version, SessionStateUpdateCriteria& session_uc) {
+  current_version_                      = new_session_version;
+  session_uc.is_current_version_updated = true;
+  session_uc.updated_current_version    = new_session_version;
   MLOG(MINFO) << " Current version is " << get_current_version();
 }
+
 /* Add PDR rule to this rules session list */
-void SessionState::insert_pdr(SetGroupPDR* rule) {
-  PdrList_.push_back(*rule);
+void SessionState::insert_pdr(
+    SetGroupPDR* rule, SessionStateUpdateCriteria& session_uc, bool crit_add) {
+  // Check if it already exists
+  int32_t Pdr_index;
+  Pdr_index = pdr_index(rule->pdr_id());
```

Comment:
is `pdr_index` a constructor? if it's a function let's rename it to `get_pdr_index`

Reply:
Yes. Fixed it. Sorry for wrong naming conversion.

Reply:
no worries :) Thanks for fixing.